### PR TITLE
Fix TxOut::is_fee, add confidential is_explicit and unwrap_explicit, add Transaction::fee

### PR DESCRIPTION
**Pull request by stevenroose**

### CODE REVIEW

**Comment by RCasatta** [comment]
`o.minimum_value()` is used to get the Explicit Value, which is always the case for fee, so it works but it was unclear to me by looking at this function.

**Comment by stevenroose** [comment]
@RCasatta I agree, I had to check it as well. This is how it was done in our integration tests, I ported the method here. I guess since we use is_fee, we can assert the value to be explicit afterwards. Let me clean that up and add a test.

**Comment by stevenroose** [comment]
I noticed that is_fee was actually incomplete according to Elements Core. So I fixed that and added is_explicit and unwrap_explicit methods to the confidential types to help.

**Comment by RCasatta** [comment]
did you consider having `unwrap_explicit` returning a Result? This way you can remove the panic! and also remove `is_explicit` and use `unwrap_explicit().is_ok()` instead

**Comment by stevenroose** [comment]
Yeah but then it's not really an unwrap method like in the stdlib. We could have `fn explicit(&self) -> Option<T>` instead, like [`Result::ok`](https://doc.rust-lang.org/std/result/enum.Result.html#method.ok); in that case our usage would be `value.explicit().unwrap()`, which for me is fine as well. In that case, the `is_explicit` method could just return `.explicit().is_some()`.

**Comment by RCasatta** [comment]
yeah, Option or like it is now is ok

**Comment by stevenroose** [comment]
I changed it to a Option<u64>, can you re-review and ack? I'd like to have this to add feerates to hal-elements.

File: src/confidential.rs [hunk 1]
```diff
@@ -182,6 +182,23 @@ impl Value {
...
+    /// Check if the value is explicit.
+    pub fn is_explicit(&self) -> bool {
+        match *self {
+            Value::Explicit(_) => true,
+            _ => false,
+        }
+    }
+
+    /// Unwrap the explicit value of this type.
+    /// Panics if [is_explicit] returns false.
```
Comment (RCasatta) [inline src/confidential.rs]: fn does not panic if it is not explicit it returns None

Comment (stevenroose) [reply src/confidential.rs]: ah damn missed that comment, thanks

fixed

**Comment by RCasatta** [comment]
ack cbb9674

**Comment by stevenroose** [comment]
@apoelstra could you perhaps give me owner permissions on this repo so that I can merge PRs that are ACKed, but not "github-approved"? Currently I can't. I'll try self-approve, but that's weird.

**Comment by stevenroose** [comment]
update: you can't self-approve. Bravo, GitHub :)

**Comment by shesek** [comment]
Is there a reason `explicit()` was added to `Value` but not to `Asset` or `Nonce`, while `unwrap_explicit()` was not added to `Value` but only to the others?

I was refactoring some of my code to take advantage of the new methods, was kind of surprised to see that `Asset::explicit()` does not exists. I think it would make sense to have all three methods on all three structs.

**Comment by shesek** [comment]
Nice, @stevenroose unified them here: https://github.com/ElementsProject/rust-elements/commit/5d826e8d1c21a8bc849f7019fab876799cbc9ecd

**Pull request merged**